### DISCUSSION TITLE
cvodes integrator

### DISCUSSION THREAD

**Discussion by PrakitrSrisuma** [casadi/casadi#3096] [Q&A]
Hello,

I would like to know how to specify a time-dependent parameter into the integrator cvodes.
For example

   dae = struct('x',x,'p',p,'ode',xdot);
   F = integrator('F', 'cvodes', dae, 0, tf);

In this case, the "p" is a linear function of time, e.g., p = at+b, and so I would like cvodes to consider this when integrating the ODE. I am not sure what is the correct way to specific this.

**Comment by brunoml5** [comment]
Hi!

I did the following test and it worked. It appears that the 'integrator' function accepts 't' as an input.

`T = 10. # Time horizon
N = 20 # number of control intervals

x1 = MX.sym('x1')
x2 = MX.sym('x2')
x = vertcat(x1, x2)
u = MX.sym('u')

t = MX.sym('t')

xdot = vertcat((1-x2**2)*x1 - x2 + u, x1 + t)

L = t

dae = {'x':x, 'p':u, 't':t, 'ode':xdot, 'quad':L}
F_t = integrator('F_t', 'cvodes', dae, 0, T/N)

Fk = F_t(x0=[0.2,0.3], p=0.4)
print(Fk['xf'])
print(Fk['qf'])`

You can see that the output for 'qf' is (t^2)/2 = (0.5^2)/2 = 0.125.

Nevertheless, in your model, do some tests to see if this functionality is working properly

**Reply by PrakitrSrisuma** [reply]
Thank you! I will try that.

**Comment by jaeandersson** [comment]
> Hello,
> 
> I would like to know how to specify a time-dependent parameter into the integrator cvodes. For example
> 
> dae = struct('x',x,'p',p,'ode',xdot); F = integrator('F', 'cvodes', dae, 0, tf);
> 
> In this case, the "p" is a linear function of time, e.g., p = at+b, and so I would like cvodes to consider this when integrating the ODE. I am not sure what is the correct way to specific this.

Create a state $x_{\text{new}}$ defined by: $x_{\text{new}}(0) = b$ and $\dot{x}_{\text{new}}(t) = a$. You need to treat $a$ as a parameter of your ODE:
```matlab
dae = struct('x',[x; p],'p',a,'ode',[xdot; a]);
F = integrator('F', 'cvodes', dae, 0, tf);
```

**Reply by PrakitrSrisuma** [reply]
Thank you very much! Will check it out.
